### DISCUSSION TITLE
Thoughts on integrating Gitpod and Github Codespaces?

### DISCUSSION THREAD

**Discussion by daawaan4x** [directus/directus#16464] [Feature Requests]
I guess it would be nice to have a one-click development environment to quickly run the repo from source 🚀 wdyt?

**Comment by rijkvanzanten** [comment]
I'd rather look into how to offer GitHub enough hints to enable that workflow from their integrated editor (hit `.` from the root of the repo)

**Reply by daawaan4x** [reply]
Forgot about github codespaces my bad, just recently learned it is now available to everyone (previously being exclusive only to teams/orgs) and now has a free tier. Github already seamlessly integrates with codespaces tho, both from the repository homepage and the integrated editor. The only missing thing left would be a `devcontainer.json` to configure.

<div align="middle">
<img height="219" src="https://user-images.githubusercontent.com/53972232/202351103-435143ae-cd27-45fc-8299-9a993ee887ec.png" alt="image"/>

<img src="https://user-images.githubusercontent.com/53972232/202351237-5520ef50-6447-457e-bb4e-265ad115433a.png" alt="image"/>
</div>

---

On a related note, how about providing both options tho since both have different pricing models and workspace specs[^1] ? The `docker-compose.yaml` file can be reused for both configurations. 
 - Github Codespaces Pricing: https://github.com/features/codespaces#pricing
 - Gitpod Pricing: https://www.gitpod.io/pricing

[^1]: I personally use gitpod since I find it cheaper than github codespaces with its unlimited hours of usage at the cost of workspaces having shared CPUs and a fixed amount of cores and RAM, but codespaces would come in handy for those who would want to spin up dedicated workspaces with higher end specs

**Reply by rijkvanzanten** [reply]
It becomes a bit of a maintenance burden on the core team. We're all developing locally, so this would mostly be as a service for external contributors to be able to spin up a copy on an online environment.

**Reply by jdamboeck** [reply]
The devcontainer.json would enable local development with the dev containers extension in vscode too.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
